src/uct/api/v2/uct_v2.h

```diff
@@ -93,9 +93,22 @@ enum uct_perf_attr_field {
     UCT_PERF_ATTR_FIELD_LATENCY            = UCS_BIT(9),
 
     /** Enable @ref uct_perf_attr_t::max_inflight_eps */
-    UCT_PERF_ATTR_FIELD_MAX_INFLIGHT_EPS   = UCS_BIT(10)
+    UCT_PERF_ATTR_FIELD_MAX_INFLIGHT_EPS   = UCS_BIT(10),
+
+    /** Enable @ref uct_perf_attr_t::flags */
+    UCT_PERF_ATTR_FIELD_FLAGS              = UCS_BIT(11)
 };
 
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Flags of supported performance attributes functionalities
+ *
+ * This is used in @ref uct_perf_attr_t::flags.
+ */
+typedef enum {
+    /** TX operations can depend on unrelated RX operation completion */
+    UCT_PERF_ATTR_FLAGS_TX_RX_DEPENDENCY = UCS_BIT(0),
```

Comment:
I'm not sure I undestand the description. Can you please elaborate how the flag will be used ?

Reply:
In case of bidirectional communications for RNDV, RDMA_READ triggered from a received RTS, can fill QP/window (because of internal FW read split), which prevents unrelated SEND (RTS) to be sent and remote RDMA_READ to be triggered. This is effectively introducing a dependency between the two unrelated flow directions. The flags intends to signal that in the broadest sense.

The flag can be used to instantiate a separate lane so that RX (RDMA_READ) will not be carried by the same QP than SEND operations.

Reply:
Is the goal to have separate TX/RX QPS ? Can just put shared RX/TX ?

Reply:
It is to separate rma lanes on their own qps so they don't interfere with am sending (which for rndv delays the remote to post its own rdma_read). For naming do you mean `UCT_PERF_ATTR_FLAGS_SHARED_RX_TX`? Seems the current one might match better?